isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -2586,7 +2586,7 @@ namespace isobus
 				{
 					if (AcknowledgementType::Negative == static_cast<AcknowledgementType>(message->get_uint8_at(0)))
 					{
-						std::uint32_t targetParameterGroupNumber = message->get_uint32_at(5, CANMessage::ByteFormat::BigEndian);
+						std::uint32_t targetParameterGroupNumber = message->get_uint24_at(5);
```

Comment:
```suggestion
						std::uint32_t targetParameterGroupNumber = message->get_uint24_at(5, CANMessage::ByteFormat::BigEndian);
```

Reply:
I think little endian is correct? The PGN is encoded like this:
```
buffer[5] = static_cast<std::uint8_t>(parameterGroupNumber & 0xFF);
buffer[6] = static_cast<std::uint8_t>((parameterGroupNumber >> 8) & 0xFF);
buffer[7] = static_cast<std::uint8_t>((parameterGroupNumber >> 16) & 0xFF);
```

Reply:
Oh huh, it was in big endian before I think? Maybe that was a bug aswell? https://github.com/ad3154/ISO11783-CAN-Stack/blob/22094dd22a460e46d8778d9a40c26ac5d529a3d0/isobus/src/isobus_virtual_terminal_client.cpp#L2591-L2593

Reply:
Oh nvm I can't read, it has the `<<` instead of `>>` haha

Reply:
> ```
> buffer[5] = static_cast<std::uint8_t>(parameterGroupNumber & 0xFF);
> buffer[6] = static_cast<std::uint8_t>((parameterGroupNumber >> 8) & 0xFF);
> buffer[7] = static_cast<std::uint8_t>((parameterGroupNumber >> 16) & 0xFF);
> ```

Huh I'm totally confused now. Isn't the little endian is the `<<`?

Reply:
> Huh I'm totally confused now. Isn't the little endian the `<<`?
Okay nvm again. That was the writing part of course, whereas we are reading it now so should be good

Reply:
Haha, no worries 🙂